packages/ComponentLibrary/src/components/IconButton/types.ts

```diff
@@ -1,10 +1,10 @@
-import { IconButtonProps, SxProps, Theme } from '@mui/material';
+import { IconButtonProps } from '@mui/material';
 
-export interface IIconComponentProps extends IconButtonProps {
+export interface IIconComponentProps extends Omit<IconButtonProps, 'children'> {
   fill?: string;
   hoverFill?: string;
   width?: number;
   height?: number;
-  tooltip?: string;
-  sx?: SxProps<Theme>;
+  tooltip: string;
+  children: React.ReactNode;
```

Comment:
Instead of manually declaring `children` prop you could use the `React.PropsWithChildren` interface and remove this line

To accomplish it, you would have to start this declaration with:
```ts
export interface IIconComponentProps extends React.PropsWithChildren<Omit<IconButtonProps, 'children'>>

```

Reply:
Great suggestion